library/common-library/DEVELOPERS.md

```diff
@@ -0,0 +1,615 @@
+URL and use `newrelic.common.images.image` instead, but it is there to be used in case it is needed.
+
+Usage:
+```mustache
+{{- /* For jobImage */}}
+{{ include "newrelic.common.images.tag" ( dict "imageRoot" .Values.jobImage "context" .) }}
+{{- /* For the integration */}}
+{{ include "newrelic.common.images.tag" ( dict "imageRoot" .Values.images.integration "context" .) }}
+{{- /* For the agent */}}
+{{ include "newrelic.common.images.tag" ( dict "imageRoot" .Values.images.agent "context" .) }}
+```
+
+### `newrelic.common.images.renderPullSecrets`
+If returns a merged map that contains the pull secrets from the global configuration and the local one.  
+
+Usage:
+```mustache
+{{- /* For jobImage */}}
+{{ include "newrelic.common.images.renderPullSecrets" ( dict "pullSecrets" .Values.jobImage.pullSecrets "context" .) }}
+{{- /* For the integration */}}
+{{ include "newrelic.common.images.renderPullSecrets" ( dict "pullSecrets" .Values.images.pullSecrets "context" .) }}
+{{- /* For the agent */}}
+{{ include "newrelic.common.images.renderPullSecrets" ( dict "pullSecrets" .Values.images.pullSecrets "context" .) }}
+```
+
+
+
+## _serviceaccount.tpl
+These functions are used to evaluate if the service account should be created, with which name and add annotations to it.
+
+The functions that the common library has implemented for serve account are:
+* `newrelic.common.serviceAccount.create`
+* `newrelic.common.serviceAccount.name`
+* `newrelic.common.serviceAccount.annotations`
+
+Usage:
+```mustache
+{{- if include "newrelic.common.serviceAccount.create" . -}}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  {{- with (include "newrelic.common.serviceAccount.annotations" .) }}
+  annotations:
+    {{- . | nindent 4 }}
+  {{- end }}
+  labels:
+    {{- include "newrelic.common.labels" . | nindent 4 }}
+  name: {{ include "newrelic.common.serviceAccount.name" . }}
+  namespace: {{ .Release.Namespace }}
+{{- end }}
+```
+
+
+
+## _affinity.tpl, _nodeselector.tpl and _tolerations.tpl
+There three files are almost equal, follow the idiomatic way that `helm create` creates by
+default. It also looks if there is a global value like other helpers.
+```yaml
+global:
+  affinity: {}
+  nodeSelector: {}
+  tolerations: []
+affinity: {}
+nodeSelector: {}
+tolerations: []
+```
+
+The values here are replaced. if a value at root level is found, the global one is ignored.
+
+Usage (example in a pod spec):
+```mustache
+spec:
+  {{- with include "newrelic.common.nodeSelector" . }}
+  nodeSelector:
+    {{- . | nindent 4 }}
+  {{- end }}
+  {{- with include "newrelic.common.affinity" . }}
+  affinity:
+    {{- . | nindent 4 }}
+  {{- end }}
+  {{- with include "newrelic.common.tolerations" . }}
+  tolerations:
+    {{- . | nindent 4 }}
+  {{- end }}
+```
+
+
+
+## _agent-config.tpl
+### `newrelic.common.agentConfig.defaults`
+This returns a YAML that the agent can use directly as a config that includes other options from the values file li verbose mode,
```

Comment:
`li`? Is it an abbreviation like i.e.?

Reply:
"like"

Reply:
Fixed in [#6216d9f](https://github.com/newrelic/helm-charts/pull/797/commits/6216d9f3f09fedf3717c1b7a457a887fd8230371)